pkg/operator/controller/dns_status_test.go

```diff
@@ -224,122 +229,80 @@ func TestComputeDNSDegradedCondition(t *testing.T) {
 		expected     operatorv1.ConditionStatus
 	}{
 		{
-			name:         "0 available, DNS invalid MaxUnavailable",
+			name:         "0 available",
 			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 0, intstr.FromString("TEST")),
-			nrDaemonset:  makeDaemonSet(6, 0, intstr.FromString("33%")),
+			dnsDaemonset: makeDaemonSet(6, 0),
+			nrDaemonset:  makeDaemonSet(6, 0),
 			expected:     operatorv1.ConditionTrue,
 		},
-		{
-			name:         "node-resolver invalid MaxUnavailable is ok",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(10, 9, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(6, 0, intstr.FromString("TEST")),
-			expected:     operatorv1.ConditionFalse,
-		},
-		{
-			name:         "DNS invalid MaxUnavailable (string with digits without a percent sign)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.IntOrString{Type: intstr.String, StrVal: "10"}),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromString("33%")),
-			expected:     operatorv1.ConditionUnknown,
-		},
-		{
-			name:         "node-resolver invalid MaxUnavailable (string with digits without a percent sign) is ok",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.IntOrString{Type: intstr.String, StrVal: "33"}),
-			expected:     operatorv1.ConditionFalse,
-		},
-		{
-			name:         "DNS invalid MaxUnavailable (string with letters)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.IntOrString{Type: intstr.String, StrVal: "TEST"}),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromString("33%")),
-			expected:     operatorv1.ConditionUnknown,
-		},
-		{
-			name:         "node-resolver invalid MaxUnavailable (string with letters) is ok",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.IntOrString{Type: intstr.String, StrVal: "TEST"}),
-			expected:     operatorv1.ConditionFalse,
-		},
```

Comment:
It's interesting that we still had these test cases for invalid `maxUnavailable` values for the node-resolver daemonset.  We kept these test cases when we changed the status reporting to ignore the node-resolver daemon set in #273.  I would guess the reasoning was that we wanted the test cases to verify that the status reporting code really was not basing status on the daemon set's`maxUnavailable` parameter.  Does the same reasoning still apply?  If so, it would make sense to keep (some of) these test cases.

Reply:
It seems to be an oversight that I didn't clean these units tests then.  We entirely removed the checking of node resolver daemonset and its `maxUnavailable` parameter. It doesn't make sense anymore to check anything having to do with node resolver when computing the degraded condition: https://github.com/openshift/cluster-dns-operator/pull/273/files#diff-32495132facf7e0819a407af732514958b198d9e40236656bc43b093345d1539L111

Reply:
In fact, I should remove the `nrDaemonset` field from the test cases.  Does that make more sense?

Reply:
> It seems to be an oversight that I didn't clean these units tests then. We entirely removed the checking of node resolver daemonset and its `maxUnavailable` parameter. It doesn't make sense anymore to check anything having to do with node resolver when computing the degraded condition

Are you sure it was an oversight?  My speculation was that it was intentional:  If parameter X *did* affect result Y, and you are changing the logic so that X *doesn't* affect Y, then it _does_ make sense for tests to verify that X doesn't affect Y, right?  More generally, I'd rather err on the side of keeping possibly redundant test cases over risking removing test cases that could possibly detect faults.  However, I'll defer to your judgment if you decide these test cases really should be removed and the test simplified.

Reply:
I am quite sure it was an oversight.  I removed the parameters `haveNodeResolverDaemonset` and `nodeResolverDaemonset` from the signature of `computeDNSDegradedCondition`, because we no longer use them in the function.  We still use them in `computeDNSProgressingCondition`, but not `computeDNSDegradedCondition`.

Reply:
All right, I can live with that.  Test coverage is still good overall.